engine/lib/filestore.php

```diff
@@ -463,11 +472,35 @@ function elgg_get_file_simple_type($mime_type) {
  * @access private
  */
 function _elgg_filestore_init() {
-	global $CONFIG;
-
-	// Now register a default filestore
-	if (isset($CONFIG->dataroot)) {
-		set_default_filestore(new \ElggDiskFilestore($CONFIG->dataroot));
+	$config = _elgg_services()->config;
+	
+	// register core fs adapters
+	$adapters = _elgg_services()->dataStorageAdapters;
```

Comment:
I imagine all this logic can go into the dataSorageAdapters di factory function. Then we don't need elgg_filestore_init at all.